src/webgl/GeometryBuilder.js

```diff
@@ -0,0 +1,139 @@
+import p5 from '../core/main';
+import * as constants from '../core/constants';
+
+/**
+ * @private
+ * A class responsible for converting successive WebGL draw calls into a single
+ * `p5.Geometry` that can be reused and drawn with `model()`.
+ */
+class GeometryBuilder {
+  constructor(renderer) {
+    this.renderer = renderer;
+    renderer._pInst.push();
+    this.identityMatrix = new p5.Matrix();
+    renderer.uMVMatrix = new p5.Matrix();
+    this.geometry = new p5.Geometry();
+    this.geometry.gid = `_p5_GeometryBuilder_${GeometryBuilder.nextGeometryId}`;
+    GeometryBuilder.nextGeometryId++;
```

Comment:
This jumps out as an unusual pattern to me. Typically don't we store data like this on the renderer object?

Reply:
In the earlier library implementation of this, I made people manually create an id, but that seemed a little awkward and easy to forget about and make duplicates. We could also use a uuid, so we randomly generate a string id. It's probably slower than just incrementing a number, but since we likely aren't creating new geometry every frame maybe that's ok and preferable to this global numeric id?

Reply:
oh also I forgot to address why it's global and not in the renderer: this is because you can draw a model to a different graphic, which is its own renderer, but we still don't want id clashes.

Reply:
I guess you could consider it a namespaced global property? Or a static class property (if we change some transpiler settings we could actually declare it as `static` within the class definition, but I got a linter error with the current settings when I first tried that.)

Reply:
> oh also I forgot to address why it's global and not in the renderer: this is because you can draw a model to a different graphic, which is its own renderer, but we still don't want id clashes.

That makes total sense, thanks for clearing that up!